ray_lightning/launchers/ray_horovod_launcher.py

```diff
@@ -0,0 +1,251 @@
+        """Return whether the launcher is interactive compatible."""
+        return True
+
+    def launch(self,
+               function: Callable,
+               *args: Any,
+               trainer: Optional["pl.Trainer"] = None,
+               **kwargs: Any) -> Any:
+        """Launch the function on the workers and collect the results."""
+        ray_output = self.run_function_on_workers(
+            function, *args, trainer=trainer, **kwargs)
+
+        if trainer is None:
+            raise NotImplementedError(
+                "Ray launcher does not support trainer is None!")
+        self._recover_results_in_main_process(ray_output, trainer)
+        return_value = ray_output.trainer_results
+
+        return return_value
+
+    def run_function_on_workers(self,
+                                function: Callable,
+                                *args: Any,
+                                trainer: Optional["pl.Trainer"] = None,
+                                **kwargs: Any):
+        """Run the function on the workers and collect the results.
+           `executor.run_remote` is used to launch multiple ray remote tasks
+            to distributed training the model using the horovod backend.
+        """
+
+        # put the model as the ray object
+        # this reduce the memory comsumption
+        # and remove the model temporarily from the args
+        model = trainer.model
+        model_ref = ray.put(model)
+        trainer.model = None
+        new_args = tuple([None] + list(args[1:]))
+
+        # remove the executor temporarily from the args
+        # in order to avoid the ray.get() call in the function
+        # because executor is not pickleable
+        executor = self._executor
+        self._executor = None
+        self._strategy.executor = None
+
+        executor.start(executable_cls=get_executable_cls())
+
+        if TUNE_INSTALLED and is_session_enabled():
+            # Create communication queue and send to all the workers.
+            self.tune_queue = Queue(actor_options={"num_cpus": 0})
+
+        self._futures = executor.run_remote(lambda: self._wrapping_function(
+            function, model_ref, new_args, kwargs, self.tune_queue))
+
+        # put back the executor and model
+        self._executor = executor
+        self._strategy.executor = executor
+        trainer.model = model
+
+        results = process_results(self._futures, self.tune_queue)
+        executor.shutdown()
+        self._strategy.teardown()
+
+        return results[0]
+
+    def _wrapping_function(
+            self,
+            function: Callable,
+            model_ref: Any,
+            args: Any,
+            kwargs: Any,
+            tune_queue: Queue,
+    ) -> Any:
+        """Wrapping function to run the function on the workers.
+            `_wrapping_function` is run on each remote worker.
+            `function(*args, **kwargs)` is where the actual training happens.
```

Comment:
Make first sentence of docstring 1 line please!

Reply:
dome